### PR TITLE
Feature/add update hook

### PR DESCRIPTION
**Pull request by mronus**
# Description 

* Add support for updating hooks
* Fix the type of parameter in `getRepositoryById`
* Replace `position` with `line` in `createReviewComment` in order to avoid position computation

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GitHub.java [hunk 1]
```diff
@@ -558,7 +558,7 @@ public GHRepository getRepository(String name) throws IOException {
      * @throws IOException
      *             the io exception
      */
-    public GHRepository getRepositoryById(String id) throws IOException {
+    public GHRepository getRepositoryById(long id) throws IOException {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GitHub.java:561]: Please make this a separate PR and keep the existing method but make it deprecated. 
Also, add tests for both the old and new methods.  Thanks!

File: src/main/java/org/kohsuke/github/GHPullRequest.java [hunk 2]
```diff
@@ -510,7 +510,7 @@ public GHPullRequestReviewComment createReviewComment(String body, String sha, S
                 .with("body", body)
                 .with("commit_id", sha)
                 .with("path", path)
-                .with("position", position)
+                .with("line", position)
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPullRequest.java]: This too needs to be a separate PR, with tests for the change.

The docs indicate that `position` and `line` are not equivalent. You'll probably need to implement a builder to support the various parameter combinations.
https://docs.github.com/en/free-pro-team@latest/rest/reference/pulls#create-a-review-comment-for-a-pull-request

File: src/main/java/org/kohsuke/github/GHHooks.java [hunk 3]
```diff
@@ -77,7 +77,7 @@ public GHHook createHook(String name, Map<String, String> config, Collection<GHE
             }
 
             GHHook hook = root.createRequest()
-                    .method("POST")
+                    .method("PATCH")
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHHooks.java]: Does `POST` not work?

https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#create-a-repository-webhook

**Review by bitwiseman — CHANGES_REQUESTED** [review]
Thank you for choosing to contribute!  
Please split these three independent changes into separate PRs and add tests.  Thanks!

**Comment by bitwiseman** [comment]
@mronus  
Do you have time to finish working on this?

**Comment by mronus** [comment]
@bitwiseman I don't know when I can finish it but will try to do it asap, like in a few weeks at the latest.

**Comment by mronus** [comment]
This PR got closed as it was split into 3 different PRs.

**Pull request closed without merging**